Code/Framework/AzToolsFramework/AzToolsFramework/Prefab/PrefabPublicHandler.cpp

```diff
@@ -719,26 +727,51 @@ namespace AzToolsFramework
                 PrefabDom parentEntityDomAfterAddingEntity;
                 m_instanceToTemplateInterface->GenerateDomForEntity(parentEntityDomAfterAddingEntity, *parentEntity);
 
-                // Create undo node to account for changes to parent entity due to adding a new entity under it. Currently only the
-                // EditorEntitySortComponent get modified on the parent but more things can change in the future too.
-                PrefabUndoHelpers::UpdateEntity(
-                    *parentEntityDomInOwningTemplate, parentEntityDomAfterAddingEntity, parentId, undoBatch.GetUndoBatch());
-            }
+                // Add entity DOM in owning template.
+                PrefabDom newEntityDom;
+                m_instanceToTemplateInterface->GenerateDomForEntity(newEntityDom, *entity);
 
-            // Select the new entity (and deselect others).
-            AzToolsFramework::EntityIdList selection = {entityId};
+                TemplateId targetTemplateId;
+                PrefabUndoAddEntity::NewEntityInfo newEntityInfo(entityId, newEntityDom);
+                PrefabUndoAddEntity::ParentEntityInfo parentEntityInfo(parentId, *parentEntityDomInOwningTemplate, parentEntityDomAfterAddingEntity);
```

Comment:
Just be careful about `parentEntityDomInOwningTemplate` here. It is a reference to template DOM. The new workflow attempts to use it after lines of code.

So, I think one way is that we can remove the scope and make a copy out of it.